src/bin/scraper.rs

```diff
@@ -83,27 +83,34 @@ fn persist<P: AsRef<Path>>(filename: P, sections: &Vec<Section>) -> Result<()> {
 
 /// Store sessions to database `conn`, creating tables and writing rows. Writing to a non-empty
 /// database will likely produce an error.
-pub fn store_sections(conn: &Connection, sections: &Vec<Section>) -> Result<()> {
+pub fn store_sections(conn: &Connection, courses: &Vec<Course>) -> Result<()> {
     conn.execute_batch(
         "PRAGMA foreign_keys = ON;
 
+        CREATE TABLE course (
+            subject_code TEXT NOT NULL,
+            course_code TEXT NOT NULL,
```

Comment:
we can get around the concatenation issue by adding `identifier` column on db creation which holds `${subject_code}${course_code}`, then create an index on it

Reply:
Meh, I don't really see it as an issue personally, I load tested `/term/202501/search` and it comes out to sub 1ms at least 95% of the time:
```console
$ echo -e "POST http://127.0.0.1:8080/term/202501/search\nContent-Type: application/x-www-form-urlencoded\n@search.test" | vegeta attack -duration=10s -rate=7500 -max-workers=1000 | tee results.bin | vegeta report
Requests      [total, rate, throughput]         75001, 7500.09, 7499.64
Duration      [total, attack, wait]             10.001s, 10s, 598.5µs
Latencies     [min, mean, 50, 90, 95, 99, max]  297.85µs, 520.195µs, 441.073µs, 732.74µs, 857.817µs, 1.289ms, 7.294ms
Bytes In      [total, mean]                     825011, 11.00
Bytes Out     [total, mean]                     975013, 13.00
Success       [ratio]                           100.00%
Status Codes  [code:count]                      200:75001  
```
My personal philosophy towards the SQLite database is that it should only store the course info in some pure form, and we should delegate performance concerns like creating a search index to the backend.